habitat/sims/pyrobot/pyrobot.py

```diff
@@ -0,0 +1,178 @@
+
+        super().__init__(config=config)
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        return spaces.Box(
+            low=self.min_depth_value,
+            high=self.max_depth_value,
+            shape=(self.config.HEIGHT, self.config.WIDTH, 1),
+            dtype=np.float32,
+        )
+
+    def get_observation(self, robot_obs):
+        obs = robot_obs.get(self.uuid, None)
+
+        assert obs is not None, "Invalid observation for {} sensor".format(
+            self.uuid
+        )
+
+        if obs.shape != self.observation_space.shape:
+            obs = cv2.resize(
+                obs,
+                (
+                    self.observation_space.shape[1],
+                    self.observation_space.shape[0],
+                ),
+            )
+
+        obs = np.clip(obs, self.config.MIN_DEPTH, self.config.MAX_DEPTH)
+        if self.config.NORMALIZE_DEPTH:
+            # normalize depth observations to [0, 1]
+            obs = (obs - self.config.MIN_DEPTH) / self.config.MAX_DEPTH
+
+        obs = np.expand_dims(obs, axis=2)  # make depth observations a 3D array
+
+        return obs
+
+
+@registry.register_simulator(name="PyRobot-v0")
+class PyRobot(Simulator):
+    def __init__(self, config: Config) -> None:
+        self._config = config
+
+        robot_sensors = []
+        for sensor_name in self._config.SENSORS:
+            sensor_cfg = getattr(self._config, sensor_name)
+            sensor_type = registry.get_sensor(sensor_cfg.TYPE)
+
+            assert sensor_type is not None, "invalid sensor type {}".format(
+                sensor_cfg.TYPE
+            )
+            robot_sensors.append(sensor_type(sensor_cfg))
+        self._sensor_suite = SensorSuite(robot_sensors)
+
+        config_pyrobot = {"base_controller": self._config.BASE_CONTROLLER}
+
+        assert (
+            self._config.ROBOT in self._config.ROBOTS
+        ), "Invalid robot type {}".format(self._config.ROBOT)
+        self._robot_config = getattr(self._config, self._config.ROBOT.upper())
+
+        self._robot = pyrobot.Robot(
+            self._config.ROBOT, base_config=config_pyrobot
+        )
+
+    def _degree_to_radian(self, degrees):
+        return (degrees / 180) * np.pi
+
+    def _robot_obs(self):
+        return {
+            "rgb": self._robot.camera.get_rgb(),
+            "depth": self._robot.camera.get_depth(),
+        }
+
+    @property
+    def sensor_suite(self) -> SensorSuite:
+        return self._sensor_suite
+
+    @property
+    def base(self):
+        return self._robot.base
+
+    @property
+    def camera(self):
+        return self._robot.camera
+
+    # TODO(akadian): add action space support.
+
+    def reset(self):
+        self._robot.camera.reset()
+
+        observations = self._sensor_suite.get_observations(self._robot_obs())
+        return observations
+
+    def step(self, action, action_params):
+        if action in self._robot_config.BASE_ACTIONS:
```

Comment:
This technique of checking actions doesn't seem extensible. What if a new type of action comes in tomorrow? Will you add if check if you add one more action type?

Reply:
That's correct, @kalyanvasudev is there a way I can get a list of supported actions in PyRobot sot that we can avoid duplication of action definition on the habitat side, I tried having a look around but did not find anything.

Reply:
No, there is no method that would return the available list of actions. I think what you have at the moment is fine or you could just change it to ```try: and except:``` routine.